dlk/tests/test_optimizer.py

```diff
@@ -13,888 +13,497 @@
-            'aq_const1',
-            Float32(),
-            np.array(1)
-        )
-
-        s2 = Constant(
-            'aq_const2',
-            Float32(),
-            np.array(2)
-        )
-
-        aq = QTZ_linear_mid_tread_half(
-            'aqtz1',
-            [1, 4, 4, 3],
-            Float32(),
-            {'X': i2, 'Y': s1, 'Z': s2}
-        )
-
-        w2 = Constant(
-            'weight2',
-            Float32(),
-            data2
-        )
-
-        q2 = QTZ_binary_mean_scaling(
-            'qtz2',
-            [3, 2, 2, 3],
-            Float32(),
-            {'input': w2}
-        )
-
-        conv2 = Conv(
-            'conv2',
-            [1, 3, 3, 3],
-            Float32(),
-            {'X': aq, 'W': q2},
-            kernel_shape=[2, 2]
-        )
-
-        w3 = Constant(
-            'weight3',
-            Float32(),
-            data3
-        )
-
-        q3 = QTZ_binary_mean_scaling(
-            'qtz3',
-            [3, 2, 2, 3],
-            Float32(),
-            {'input': w3}
-        )
-
-        conv3 = Conv(
-            'conv3',
-            [1, 3, 3, 3],
-            Float32(),
-            {'X': aq, 'W': q3},
-            kernel_shape=[2, 2]
-        )
-
-        y1 = Output(
-            'output1',
-            [1, 3, 3, 3],
-            Float32(),
-            {'input': conv2}
-        )
-
-        y2 = Output(
-            'output2',
-            [1, 3, 3, 3],
-            Float32(),
-            {'input': conv3}
-        )
+        # activation quantizer
+        s1 = Constant('aq_const1', Float32(), np.array(1))
+        s2 = Constant('aq_const2', Float32(), np.array(2))
+        aq = QTZ_linear_mid_tread_half('aqtz1', [1, 4, 4, 3], Float32(), {'X': conv1, 'Y': s1, 'Z': s2})
+
+        # Conv2
+        w2 = Constant('weight2', Float32(), data2)
+        kq = QTZ_binary_mean_scaling('kqtz1', [1, 2, 2, 3], Float32(), {'input': w2})
+        conv2 = Conv('conv2', [1, 3, 3, 3], Float32(), {'X': aq, 'W': kq}, kernel_shape=[2, 2])
+        conv2.a_quantizer = [aq]
+        conv2.quantizer = kq
+
+        # One output
+        y = Output('output', [1, 3, 3, 3], Float32(), {'input': conv2})
 
         # add ops to the graph
-        graph.add_op_and_inputs(y1)
-        graph.add_op_and_inputs(y2)
+        graph.add_op_and_inputs(y)
 
         return graph
 
-    def create_quantized_graph2(self, data1: np.ndarray, data2: np.ndarray, data3: np.ndarray) -> Graph:
+
+class TestPassPackWeights(unittest.TestCase):
+    """Test class for packing weight."""
+    def test_pass_pack_weights(self) -> None:
+        """Test pass."""
+        data1 = np.float32(np.random.rand(1, 2, 2, 3))
+        data2 = np.float32(np.random.rand(1, 2, 2, 3))
+        graph1 = self.create_sample_graph(data1, data2)
+
+        pass_pack_weights(graph1)
+
+        self.assertEqual(graph1.get_op('conv2').input_ops['W'].op_type, 'Constant',
+                         '[Failed] Found input kernel weights not a constant')
+
```

Comment:
It is better to add test that `graph` can't get `kqtz1` and `weight2` op.

Reply:
Thanks, new test added for the case mentioned above.

Reply:
:+1: